scripts/c_header_guards.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+"""
+Script to check/fix header guards in C header files.
+Converts guards to the format
+    TF_PSA_CRYPTO_<rel_path>_<filename>_H
+where:
+- 'TF_PSA_CRYPTO_' is the common hardcoded prefix;
+- 'rel_path' is the path used in C files to include such header file.
+
+Usage: python3 fix_header_guards.py
+"""
+
+import sys
+import os
+import re
+import argparse
+from typing import List, Tuple, Match
+
+import framework_scripts_path # pylint: disable=unused-import
+from mbedtls_framework import build_tree
+
+PREFIX = 'TF_PSA_CRYPTO_'
+
+# Dictionary containing the folders where header files are to be fixed/checked:
+# - key: path of header files to check;
+# - value: path that must be trimmed from the key value in order to get the
+#          final guard.
+FOLDERS_TO_CHECK = {
+    'drivers/builtin/include/mbedtls/': 'drivers/builtin/include/',
+    'drivers/builtin/include/mbedtls/private/': 'drivers/builtin/include/',
+    'drivers/builtin/src/': 'drivers/builtin/src/',
+    'drivers/everest/include/tf-psa-crypto/private/everest/':
+        'drivers/everest/include/tf-psa-crypto/private/everest/',
+    'drivers/p256-m/': 'drivers/p256-m/',
+    'drivers/p256-m/p256-m/': 'drivers/p256-m/',
+    'include/mbedtls/': 'include/',
+    'include/psa/': 'include/',
+    'include/tf-psa-crypto/': 'include/',
+}
+
+def generate_guard_name(file_path: str, trim_path: str) -> str:
+    """Generate the new header guard."""
+    adjusted_name = file_path
+    # Trim prefix
+    if adjusted_name.startswith(trim_path):
+        adjusted_name = adjusted_name[len(trim_path):]
+    # Remove suffix
+    if adjusted_name.endswith('.h'):
+        adjusted_name = adjusted_name[:-2]
+
+    adjusted_name = re.sub(r'[^a-zA-Z0-9]', '_', adjusted_name).upper()
+    guard_elements = []
+    # Add prefix only to non public header files
+    if file_path.startswith('drivers'):
```

Comment:
Missing private headers in `core`.

Also, if we switch to the rule suggested by Gilles, the condition should be
```py
if not (adjusted_name.startswith('MBEDTLS_') or adjusted_name.startswith('PSA_') or adjusted_name.startswith('TF_PSA_CRYPTO_')):
```
That is, we no longer need to distinguish between private and public headers, we just check if the current name already has one of our 3 prefixes or not.